### PR TITLE
Record searches made on the reader page.

### PR DESCRIPTION
**Pull request by shanear**
## Test plan
- tested locally
- used `ensure_stable` to test the automated test

connects #2561

### CODE REVIEW

File: client/app/components/reader/DocumentListHeader.jsx [hunk 1]
```diff
@@ -7,74 +7,92 @@ import { connect } from 'react-redux';
...
-export const DocumentListHeader = (props) => {
-  const buttonText = props.expandAll ? 'Collapse all' : 'Expand all';
+class DocumentListHeader extends React.Component {
+  // Record the search value for analytics purposes, don't worry if it fails.
+  recordSearch = (query) => {
+    ApiUtil.post(
+      `/reader/appeal/${this.props.vacolsId}/claims_folder_searches`,
+      { data: { query } }
+    ).then(() => {
+      return true; /* noop */
```
Comment (mdbenjam) [inline client/app/components/reader/DocumentListHeader.jsx]: You can use _.noop it's a function that doesn't do anything.

File: client/app/components/reader/DocumentListHeader.jsx [hunk 2]
```diff
@@ -7,74 +7,92 @@ import { connect } from 'react-redux';
...
-    props.docFilterCriteria.category[category]
-  );
-  const tagFilters = Object.keys(props.docFilterCriteria.tag).some((tag) =>
-    props.docFilterCriteria.tag[tag]
-  );
-  const filteredCategories = [].concat(
-    categoryFilters ? ['categories'] : [],
-    tagFilters ? ['tags'] : []).join(' ');
+  render() {
+    const props = this.props;
```
Comment (mdbenjam) [inline client/app/components/reader/DocumentListHeader.jsx]: This is just so you don't have to change everything to this.props?

File: client/app/components/reader/DocumentListHeader.jsx [hunk 3]
```diff
@@ -7,74 +7,92 @@ import { connect } from 'react-redux';
...
-        {props.numberOfDocuments} Documents
-      </div>
-      <div className="usa-width-one-third">
-        <span className="cf-right-side">
-          <Button
-            name={buttonText}
-            onClick={props.toggleExpandAll}
-            id="btn-default"
-            disabled={props.noDocuments}
+    const buttonText = props.expandAll ? 'Collapse all' : 'Expand all';
```
Comment (mdbenjam) [inline client/app/components/reader/DocumentListHeader.jsx]: Hard to tell, are there any changes in here? Looks like you just made this inside a render function and the diff algorithm is lame AF.

File: client/app/components/reader/DocumentListHeader.jsx (hunk 2 above)
Comment (shanear) [inline client/app/components/reader/DocumentListHeader.jsx]: maaaaaybe 😬.

I can change it over, but wasn't sure why we weren't destructing the props argument so decided not to bother.

File: client/app/components/reader/DocumentListHeader.jsx [hunk 4]
```diff
@@ -7,74 +7,92 @@ import { connect } from 'react-redux';
...
+
+    return <div>
+      <div className="usa-grid-full document-list-header">
+        <div className="usa-width-one-third">
+          <SearchBar
+            id="searchBar"
+            onChange={props.setSearch}
+            onClearSearch={props.clearSearch}
+            onClick={props.clickSearch}
+            recordSearch={this.recordSearch}
```
Comment (shanear) [inline client/app/components/reader/DocumentListHeader.jsx]: @mdbenjam added this line and tabbed everything over. That is all.

File: client/app/components/reader/DocumentListHeader.jsx (hunk 1 above)
Comment (shanear) [inline client/app/components/reader/DocumentListHeader.jsx]: i was wondering about this. Thx for the 1337 tip.

File: client/app/components/reader/DocumentListHeader.jsx (hunk 2 above)
Comment (shanear) [inline client/app/components/reader/DocumentListHeader.jsx]: theres also a lot of props. Didn't see the upside in destructuring them

File: client/app/components/reader/DocumentListHeader.jsx (hunk 4 above)
Comment (mdbenjam) [inline client/app/components/reader/DocumentListHeader.jsx]: cool, that's what I thought--just wanted to know if I had to look real closely

**Pull request merged**